pubtools/pulplib/_impl/model/repository/yum.py

```diff
@@ -1,9 +1,86 @@
-from .base import Repository, repo_type
+from .base import Repository, SyncOptions, repo_type
 from ..frozenlist import FrozenList
 from ..attr import pulp_attrib
 from ... import compat_attr as attr
 
 
+@attr.s(kw_only=True, frozen=True)
+class YumSyncOptions(SyncOptions):
+    """Options controlling a container repository
+    :meth:`~pubtools.pulplib.YumRepository.sync`.
+    """
+
+    query_auth_token = pulp_attrib(default=None, type=str)
+    """An authorization token that will be added to every request made to the feed URL's server
+    """
+
+    validate = pulp_attrib(default=None, type=bool)
+    """If True, checksum of each file will be verified against the metadata's expectation
+    """
+
+    max_downloads = pulp_attrib(default=None, type=int)
+    """Number of threads used when synchronizing the repository.
+    """
+
+    remove_missing = pulp_attrib(default=None, type=bool)
+    """If true, as the repository is synchronized, old rpms will be removed.
+    """
+
+    retain_old_count = pulp_attrib(default=None, type=int)
+    """Count indicating how many old rpm versions to retain.
+    """
+
+    skip = pulp_attrib(default=None, type=list)
+    """List of content types to be skipped during the repository synchronization
+    """
+
+    checksum_type = pulp_attrib(default=None, type=str)
+    """checksum type to use for metadata generation.
+
+    Defaults to source checksum type of sha256
+    """
+
+    num_retries = pulp_attrib(default=None, type=int)
+    """Number of times to retry before declaring an error during repository synchronization
+
+    Default is to 2.
+    """
+
+    copy_children = pulp_attrib(default=None, type=bool)
```

Comment:
Does this actually do anything / make sense here?

I looked at the docs at https://docs.pulpproject.org/en/2.14/plugins/pulp_rpm/tech-reference/yum-plugins.html

It says "Supported only as an override config option to a repository copy command", so I suppose the option is useless when doing a sync?